build-tools/packages/version-tools/api-report/version-tools.api.md

```diff
@@ -37,6 +37,9 @@ prereleaseIndentifier: string
 // @public
 export function fromVirtualPatchScheme(virtualPatchVersion: semver.SemVer | string): semver.SemVer;
 
+// @public
+export function generateLegacyCompatRange(version: semver.SemVer | string, compatVersionInterval: number): string;
```

Comment:
You shouldn't need a new function. I think this should just be a new "maxAutomaticBump" value for the existing getVersionRange function. E.g. 

```ts
export function getVersionRange(
  version: semver.SemVer | string, 
  maxAutomaticBump: "minor" | "patch" | "~" | "^" | ["legacyCompat", 10]): string;
```

That's assuming you need to pass the compatVersionInterval (I'm not convinced that's needed). Anyway in the example above I used a tuple to tie the "legacyCompat" value to the interval. You could also use an object, but I think the tuple is a little better.

Actually, you might consider something like this:

```ts
export function getVersionRange(
  version: semver.SemVer | string, 
  maxAutomaticBump: "minor" | "patch" | "~" | "^" | number): string;
```

That signature sort of says, "give me a range that'll bump to minor/patch/etc, or if I give you a number, use that value as the compatVersionInterval - the number implies the "legacyCompat" so no need to pass both.

Reply:
The `getVersionRange` function was modified earlier and later adjusted based on the feedback here:https://github.com/microsoft/FluidFramework/pull/22687#discussion_r1783321936. I had previously made changes that aligned with your first suggestion: https://github.com/microsoft/FluidFramework/pull/22687/commits/05ba3fc23c3d40bb25f993c2345d34ce0b4ec4a4#diff-8f93c52ef80a16ba168b511033454722ba97475cef9f154257c8ebca6c0b1695, however, those changes would have conflicted with the intended purpose of the getVersionRange function.

Reply:
Ahh right, sorry for the runaround. This function should really be called getInternalVersionRanges given how it behaves.

I'd be open to updating this function to work on regular semvers too, but it's probably cleaner to leave it as it is.

But maybe this new function can live in build-cli? That way it doesn't have to be exported, and it's not _super_ related to the other stuff in version-tools.